### DISCUSSION TITLE
Mellisa: Where do I insert the "newsletter-popup__submit" class?

### DISCUSSION THREAD

**Discussion by jpbraun** [GetPublii/Publii#1892] [Help]
Hi,
I am using the Theme Mellisa. It contains a newsletter popup. 
Where do I have to add the "newsletter-popup__submit" class?

Theme -> Newsletter -> _"Remember to add a "newsletter-popup__submit" class to the submit button for the proper working of the newsletter popup."_

https://getpublii.com/docs/make-newsletter-form-functional-and-visually-appealing.html
_"Some Publii themes, especially Premium such as [Mellisa](https://marketplace.getpublii.com/product/mellisa/) or [Cortado](https://marketplace.getpublii.com/product/cortado/) include a newsletter form in the popup. In order to make the [Close] and [Submit] buttons work properly just add to the second one a "newsletter-popup__submit" class."_

Best regards,
Jan-Peter

**Comment by candidexmedia** [comment]
What platform are you using for your newsletter service?

**Reply by jpbraun** [reply]
I would like to use the new newsletter service from Hyvor Talk, because I am using the Hyvor Talk comment system.

**Reply by candidexmedia** [reply]
I don't think that class is relevant to you, since ([according to Hyvor's docs](https://talk.hyvor.com/docs/install#newsletters)), you are using a script to generate the subscription form, like this:

```html
<script async src="https://talk.hyvor.com/embed/newsletter.js" type="module"></script>
<hyvor-talk-newsletter website-id="YOUR_WEBSITE_ID"></hyvor-talk-newsletter>
```

Adding a class name to a submit button is more relevant for newsletter form embeds which make use of an unstyled HTML form with a form action, such as what is used for services like Mailchimp or [Listmonk](https://listmonk.app/):

```html
<form method="post" action="http://listmonk.com/subscription/form" class="listmonk-form">
  <div>
    <h3>Subscribe</h3>
    <input type="hidden" name="nonce" />

    <p><input type="email" name="email" required placeholder="E-mail" /></p>
    <p><input type="text" name="name" placeholder="Name (optional)" /></p>

    <p>
      <input id="e9bfe" type="checkbox" name="l" checked value="e9bfe744-a78d-4855-a16f-563d23e900fb" />
      <label for="e9bfe">Opt-in list</label>
    </p>

    <input type="submit" value="Subscribe " />
  </div>
</form>
```

In the example above, the user would add `class="newsletter-popup__submit"` in the HTML tag for the submit input, like this:

```html
<input type="submit" value="Subscribe " class="newsletter-popup__submit" />
```

See my other post for more info: https://github.com/GetPublii/Publii/discussions/1837#discussioncomment-11455376

**Comment by jpbraun** [comment]
Thank you very much. I integrated the Newsletter form.
